### PR TITLE
[AI] Update trace details and add test coverage

### LINKED ISSUE #9368: [AI][Tool][Tests] get_traces
**Description**
Validate the `get_traces` tool integration to ensure distributed tracing data is correctly exposed through the MCP server.

**Tasks**
* **Schema Verification:** Sync parameters for trace IDs, service names, and time ranges between the MCP server and Kiali API.
* **Functional Testing:** Confirm that the tool returns the expected trace spans and metadata correctly.
* **Corner Case Handling:**
    * Test behavior when the tracing backend (Jaeger/Tempo) is disabled or unreachable.
    * Handle cases where a specific Trace ID is not found.
    * Implement safety checks to avoid `panics` when handling large trace payloads or unexpected data formats.

### PR DESCRIPTION
**Pull request by josunect**
### Describe the change

- Added trace ID format validation matching the REST API's ^[a-fA-F0-9]{1,32}$ regex. Invalid IDs now return 400 instead of being passed unchecked to the tracing backend.

- Added MaxTracesLimit = 10000 constant, aligned with the REST API's cap in handlers/tracing.go.

- Added limit cap in parseArgs: if limit > MaxTracesLimit, it's clamped to 10000. This prevents arbitrarily large queries to the tracing backend.

Add test coverage: 

TestExecute_InvalidTraceIDFormat -- 4 subtests (non-hex, too long, spaces, special chars)
TestExecute_ValidTraceIDFormats -- verifies valid hex IDs pass the regex
TestExecute_BackendError_Returns503 -- mocked backend error returns 503
TestExecute_TraceNotFound_Returns404 -- nil trace returns 404
TestExecute_EmptySpansWithErrors_Returns404 -- empty spans + errors returns 404
TestExecute_ValidTrace_Returns200WithHierarchy -- full happy-path with hierarchy

TestParseArgs_LimitCap -- verifies limit is capped at 10000
TestParseArgs_LimitBelowCap -- verifies limits below cap are preserved
TestExecute_BackendError_Returns503 -- mocked connection error returns 503
TestExecute_EmptyTraces_Returns200WithZero -- empty result returns 200 with total_found: 0
TestExecute_WithTraces_Returns200WithSummary -- full happy-path with trace data
TestExecute_TracingDisabled_Returns503 -- disabled tracing returns 503

### Steps to test the PR

Recommendations for how to test this PR. Reminder that each PR should also include a "Test" label.

### Automation testing

If applicable, explain the case scencarios covered by **unit / integration / e2e** tests created for this PR.

### Issue reference

Fixes https://github.com/kiali/kiali/issues/9368

### CODE REVIEW

File: ai/mcp/get_trace_details/get_trace_details.go [hunk 1]
```diff
@@ -2,13 +2,16 @@ package get_trace_details
...
 	"net/http"
+	"regexp"
 	"strconv"
 	"strings"
 
 	"github.com/kiali/kiali/ai/mcputil"
 	jaegerModels "github.com/kiali/kiali/tracing/jaeger/model/json"
 )
 
+var validTraceIDRe = regexp.MustCompile(`^[a-fA-F0-9]{1,32}$`)
```
Comment (hhovsepy) [inline ai/mcp/get_trace_details/get_trace_details.go]: Is it possible to reuse the variable with similar name `validTraceIDRe` from `handlers/tracing.go` ?

File: ai/mcputil/defaults.go [hunk 2]
```diff
@@ -33,4 +33,5 @@ const (
 	DefaultLookbackSeconds = 600 // 10m
 	DefaultMaxSpans        = 7
 	DefaultTracesLimit     = 10
+	MaxTracesLimit         = 10000
```
Comment (hhovsepy) [inline ai/mcputil/defaults.go]: There is one more place with hardcoded value 'const maxTracingLimit = 10000' in `handlers/tracing.go` where this variable can be reused, if it makes sense of course.

File: ai/mcp/list_traces/list_traces.go [hunk 3]
```diff
@@ -143,6 +143,9 @@ func parseArgs(args map[string]interface{}, conf *config.Config) GetTracesArgs {
 	out.ErrorOnly = mcputil.AsBoolFromArgs(args, "errorOnly", "error_only")
 
 	out.Limit = mcputil.AsIntOrDefault(args, mcputil.DefaultTracesLimit, "limit")
+	if out.Limit > mcputil.MaxTracesLimit {
```
Comment (hhovsepy) [inline ai/mcp/list_traces/list_traces.go]: one suggestion here, check the `<=0` values as well

**Comment by hhovsepy** [comment]
Lgtm @josunect just a few suggestions Cursor made.

**Comment by josunect** [comment]
> Lgtm @josunect just a few suggestions Cursor made.

Thanks, @hhovsepy ! I've applied the suggestions, except the last one (Added a comment there).

File: ai/mcp/list_traces/list_traces.go (hunk 3 above)
Comment (josunect) [reply ai/mcp/list_traces/list_traces.go]: This is really needes? Negative and zero values can never reach parseArgs's output, they're caught by `AsIntOrDefault` and replaced with the default.

**Review by hhovsepy — APPROVED** [review]
Thanks @josunect, all works fine now. Sure `AsIntOrDefault` is covering negatives, my fault.

**Comment by josunect** [comment]
> Thanks @josunect, all works fine now. Sure `AsIntOrDefault` is covering negatives, my fault.

Thank you, @hhovsepy !
